### PR TITLE
Merge `jetty-11.0.x` into `jetty-12.0.x` - July 14

### PR DESCRIPTION
**Pull request by joakime**
This is a merge from `jetty-11.0.x` into `jetty-12.0.x`.
This merge has to be a PR as now branch `jetty-12.0.x` is protected from direct push.

### CODE REVIEW

File: documentation/jetty-documentation/pom.xml [hunk 1]
```diff
@@ -83,7 +83,7 @@
             <version>${project.version}</version>
             <prog-guide>../programming-guide/index.html</prog-guide>
             <op-guide>../operations-guide/index.html</op-guide>
-            <javadoc-url>https://www.eclipse.org/jetty/javadoc/jetty-11</javadoc-url>
+            <javadoc-url>https://eclipse.dev/jetty/javadoc/jetty-11</javadoc-url>
```
Comment (sbordet) [inline documentation/jetty-documentation/pom.xml]: Should be `jetty-12` at the end.

File: jetty-core/jetty-util/src/main/config/modules/console-capture.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# DO NOT EDIT - See: https://www.eclipse.org/jetty/documentation/
+# DO NOT EDIT - See: https://eclipse.dev/jetty/documentation/
```
Comment (sbordet) [inline jetty-core/jetty-util/src/main/config/modules/console-capture.mod]: Add "THIS FILE" like the others.

File: jetty-core/jetty-xml/src/test/java/org/eclipse/jetty/xml/XmlConfigurationTest.java [hunk 3]
```diff
@@ -1963,6 +1967,47 @@ public void testVirtualCallWithClassAttribute() throws Exception
...
+     */
+    @ParameterizedTest
+    @MethodSource("xmlSystemIdSource")
+    public void testSystemIdVariants(String xmlSystemId) throws IOException, SAXException
+    {
+        // empty raw xml, just to instantiate XmlConfiguration, so we can access the XmlParser / ConfigurationParser.
+        XmlConfiguration xmlConfiguration = asXmlConfiguration("<Configure class=\"org.eclipse.jetty.xml.TestConfiguration\" />");
+        XmlParser configurationProcessor = xmlConfiguration.getXmlParser();
+        XmlParser.Handler xmlHandler = configurationProcessor.newHandler();
+        InputSource inputSource = xmlHandler.resolveEntity(null, xmlSystemId);
```
Comment (sbordet) [inline jetty-core/jetty-xml/src/test/java/org/eclipse/jetty/xml/XmlConfigurationTest.java]: Can we avoid making `Handler` public, and its related getter, maybe adding `XmlParser.resolveEntity()` as package private only used by tests?

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jsp-webapp/src/main/webapp/index.jsp [hunk 4]
```diff
@@ -10,9 +10,9 @@
 
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jsp-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jsp-webapp">Source</a>
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jsp-webapp/src/main/webapp/index.jsp]: Is this change correct? It lost `ee10` references.

File: jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlParser.java [hunk 5]
```diff
@@ -350,7 +353,7 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
-    private class Handler extends DefaultHandler
+    class Handler extends DefaultHandler
```
Comment (sbordet) [inline jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlParser.java]: Make it private again.

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/webapp/login.html [hunk 6]
```diff
@@ -6,9 +6,9 @@
 <body>
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jaas-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jaas-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jaas-webapp/src/main/webapp/login.html]: Update URL path

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jetty-webapp/src/main/webapp/auth.html [hunk 7]
```diff
@@ -9,9 +9,9 @@
 
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jetty-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jetty-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jetty-webapp/src/main/webapp/auth.html]: Update URL path

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jetty-webapp/src/main/webapp/remote.html [hunk 8]
```diff
@@ -8,9 +8,9 @@
 <body>
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jetty-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jetty-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jetty-webapp/src/main/webapp/remote.html]: Update URL path

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jetty-webapp/src/main/webapp/rewrite/info.html [hunk 9]
```diff
@@ -9,9 +9,9 @@
 <body>
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jetty-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jetty-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jetty-webapp/src/main/webapp/rewrite/info.html]: Update URL path

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jaas-webapp/src/main/webapp/login.html [hunk 10]
```diff
@@ -6,9 +6,9 @@
 <body>
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jaas-webapp">Source</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jaas-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jaas-webapp/src/main/webapp/login.html]: Update URL path

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jaas-webapp/src/main/webapp/login.html [hunk 11]
```diff
@@ -6,9 +6,9 @@
...
       <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jaas-webapp/src/main/webapp/login.html]: Update URL path

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jetty-webapp/src/main/webapp/auth.html [hunk 12]
```diff
@@ -9,9 +9,9 @@
 
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jetty-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jetty-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jetty-webapp/src/main/webapp/auth.html]: Update URL path

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jetty-webapp/src/main/webapp/remote.html [hunk 13]
```diff
@@ -8,9 +8,9 @@
 <body>
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jetty-webapp">Source</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
-      <a class="menu" href="https://www.eclipse.org/jetty/documentation/current/">Documentation</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jetty-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jetty-webapp/src/main/webapp/remote.html]: Update URL path

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jetty-webapp/src/main/webapp/rewrite/info.html [hunk 14]
```diff
@@ -9,9 +9,9 @@
 <body>
     <div class="topnav">
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
-      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-11.0.x/demos/demo-jetty-webapp">Source</a>
+      <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/demos/demo-jetty-webapp">Source</a>
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-jetty-webapp/src/main/webapp/rewrite/info.html]: Update URL path

File: jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-template/src/main/resources/index.html [hunk 15]
```diff
@@ -12,7 +12,7 @@
       <!-- Change source to suit -->
       <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/jetty-eeX/jetty-eeX-demos/jetty-eeX-XXX">Source</a>
       <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-demos/jetty-ee9-demo-template/src/main/resources/index.html]: Update URL path

File: jetty-home/src/main/resources/modules/demo.d/moved/index.html [hunk 16]
```diff
@@ -10,7 +10,7 @@
       <a class="menu" href="http://localhost:8080/">Demo Home</a>
       <a class="menu" href="https://github.com/eclipse/jetty.project/tree/jetty-12.0.x/jetty-home/src/main/resources/modules/demo.d/moved">Source</a>
       <a class="menu" href="https://www.eclipse.org/jetty/">Jetty Project Home</a>
```
Comment (joakime) [inline jetty-home/src/main/resources/modules/demo.d/moved/index.html]: Update URL path

**Review by joakime — COMMENTED** [review]
More changes needed

File: jetty-ee9/jetty-ee9-jspc-maven-plugin/src/main/java/org/eclipse/jetty/ee9/jspc/plugin/JspcMojo.java [hunk 17]
```diff
@@ -57,12 +57,11 @@
...
  * jsps, which will be the Eclipse java compiler.
- * <p>
- * See <a
- * href="https://www.eclipse.org/jetty/documentation/current/jetty-jspc-maven-plugin.html">Usage
- * Guide</a> for instructions on using this plugin.
  * </p>
+ * <p>
  * Runs jspc compiler to produce .java and .class files
+ * </p>
+ * @see <a href="https://www.eclipse.org/jetty/documentation/">Usage Guide</a>
```
Comment (sbordet) [inline jetty-ee9/jetty-ee9-jspc-maven-plugin/src/main/java/org/eclipse/jetty/ee9/jspc/plugin/JspcMojo.java]: eclipse.dev.

File: jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jsp-webapp/src/main/webapp/index.jsp (hunk 4 above)
Comment (sbordet) [reply jetty-ee10/jetty-ee10-demos/jetty-ee10-demo-jsp-webapp/src/main/webapp/index.jsp]: Still missing the ee10 reference.

**Review by sbordet — CHANGES_REQUESTED** [review]

**Review by sbordet — APPROVED** [review]

**Pull request merged**